controller/linodemachine_controller_helpers.go

```diff
@@ -345,6 +377,95 @@ func getFirewallID(ctx context.Context, machineScope *scope.MachineScope, logger
 	return *linodeFirewall.Spec.FirewallID, nil
 }
 
+func getNextIP(ips []string, prefixStr string) string {
+	prefix := netip.MustParsePrefix(prefixStr)
+	currentIp := prefix.Addr().Next()
+
+	ipString := currentIp.String()
+	for {
+		if !slices.Contains(ips, ipString) {
+			break
+		}
+		currentIp = currentIp.Next()
+		ipString = currentIp.String()
+	}
+	return ipString
+}
+
+func createIPsConfigMap(ctx context.Context, machineScope *scope.MachineScope, ip string) error {
+	return machineScope.Client.Create(ctx, &corev1.ConfigMap{
```

Comment:
can you give an example of what is in this configmap? I wonder if we could use part of the LInodeCluster spec for this information instead

Reply:
it is a simple map of machineName -> IP address
```
apiVersion: v1
data:
  schinmai-test-control-plane-f4g7c: 10.0.0.3
  schinmai-test-control-plane-mfxh6: 10.0.0.2
  schinmai-test-control-plane-mhczj: 10.0.0.1
kind: ConfigMap
metadata:
  creationTimestamp: "2024-10-02T19:12:38Z"
  labels:
    clusterctl.cluster.x-k8s.io/move: "true"
  name: schinmai-test-ips
  namespace: default
  resourceVersion: "7452"
  uid: cf59bb0a-4340-4461-8317-c02748d72050
```

Reply:
I opted for a configmap so that we can have a record of what IP was assigned to which machine for updates (the linodeMachine spec doesn't have it)

Reply:
could we just add this info to the status on each of the `LinodeMachines` in this case?

Reply:
still not totally sure on the consumer of this information so I might be missing part of the context

Reply:
this info is indeed saved into the `status` of linodeMachines - When X `linodeMachines` are reconciled in parallel, the linodeMacine controller needs to pick an IP for each instance, independently. To avoid assigning the same IP to multiple machines, I'm using the configmap as a locking mechanism, and a record of the IPs assigned to each machine. IF the configmap was sucessfully updated by the reconcile loop, it guaranteees that no other machine can get that IP and we eliminate conflicts